pkg/grizzly/workflow.go

```diff
@@ -528,3 +528,33 @@ func getOutputFormat(opts Opts) (string, bool, error) {
 	}
 	return "yaml", onlySpec, nil
 }
+
+func getOnlySpec(opts Opts) (bool, string, string, error) {
+	var onlySpec bool
+	context, err := config.CurrentContext()
+	if err != nil {
+		return false, "", "", err
+	}
+	if opts.HasOnlySpec {
+		onlySpec = opts.OnlySpec
+	} else {
+		onlySpec = context.OnlySpec
+	}
+
+	if !onlySpec {
+		return false, "", "", nil
+	}
+	var kind string
+	if context.ResourceKind != "" {
+		kind = context.ResourceKind
+	} else {
+		kind = opts.ResourceKind
+	}
+	var folderUID string
+	if context.FolderUID != "" {
+		folderUID = context.FolderUID
+	} else {
+		folderUID = opts.FolderUID
+	}
+	return onlySpec, kind, folderUID, nil
+}
```

Comment:
This whole function is ugly. We have a way for envvars to override context values (partially provided by Viper). However, we have not yet established a meaningful (and simple) way for CLI arguments to override context values. Therefore, we require a hack like this to conditionally extract values from either the context or the CLI depending. There's going to be a better way to do this, but that's for a different PR.

Reply:
But yeah, another PR sounds reasonable.

Reply:
You can reduce a bit the code if you change:

```
var kind string
if context.ResourceKind != "" {
  kind = context.ResourceKind
} else {
  kind = opts.ResourceKind
}
```

with this:
```
kind := context.ResourceKind
if kind == "" {
  kind = opts.ResourceKind
}
```

And same with folderUID.

Other idea could be remove the bool return and return a typed error in [this](https://github.com/grafana/grizzly/pull/334/files#diff-1c2bc400869dd54ad3f708aebc293bb446249797cd4d64de029df22eaf5376fcR545) line.